.local/bin/scripts/projectCreate

```diff
@@ -1,100 +1,189 @@
-  echo "include_directories(include)" >> CMakeLists.txt
-  echo 'file(GLOB_RECURSE SOURCES "src/*".cc)' >> CMakeLists.txt
-  echo "add_executable($exe_name \${SOURCES})" >> CMakeLists.txt
-  echo "install(TARGETS $exe_name DESTINATION /usr/bin)" >> CMakeLists.txt
-  echo "#!/bin/sh" >> build
-  echo 'if [[ -z $1 ]]; then' >> build
-    echo 'mkdir -p bin' >> build
-    echo 'cd bin' >> build
-    echo 'cmake ..' >> build
-    echo 'make' >> build
-  echo 'elif [[ "$1" == "install" ]]; then' >> build
-    echo 'mkdir -p bin' >> build
-    echo 'cd bin' >> build
-    echo 'cmake ..' >> build
-    echo 'sudo make install' >> build
-  echo 'elif [[ "$1" == "debug" ]]; then' >> build
-    echo 'mkdir -p debug' >> build
-    echo 'cd debug' >> build
-    echo 'cmake -DCMAKE_BUILD_TYPE=Debug ..' >> build
-    echo 'make' >> build
-  echo 'elif [[ "$1" == "project" ]]; then' >> build
-    echo 'mkdir -p bin' >> build
-    echo 'cd bin' >> build
-    echo 'cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=1 ..' >> build
-    echo 'cp compile_commands.json ..' >> build
-  echo 'fi' >> build
-  chmod u+x build
-  echo 'bin/*' >> .gitignore
-  echo 'debug/*' >> .gitignore
-  echo 'compile_commands.json' >> .gitignore
-  echo '.vimspector.json' >> .gitignore
-  echo '.clangd/*' >> .gitignore
-  echo '{' >> .vimspector.json
-  echo '"configurations": {' >> .vimspector.json
-  echo '"Launch": {' >> .vimspector.json
-  echo '"adapter": "vscode-cpptools",' >> .vimspector.json
-  echo '"configuration": {' >> .vimspector.json
-  echo '"request": "launch",' >> .vimspector.json
-  echo "\"program\": \"debug/$exe_name\"," >> .vimspector.json
-  echo "\"cwd\": \"`pwd`\"," >> .vimspector.json
-  echo '"externalConsole": true,' >> .vimspector.json
-  echo '"MIMode": "gdb"' >> .vimspector.json
-  echo '}' >> .vimspector.json
-  echo '}' >> .vimspector.json
-  echo '}' >> .vimspector.json
-  echo '}' >> .vimspector.json
-  ./build project
-elif [ "$1" == "py" ] || [ "$1" = "python" ]; then
-  printf "Project name: "
-  read -r pname
-  mkdir $pname || exit
-  _pname=$(echo $pname | sed -e 's/-/_/g')
-  mkdir -p $pname/docs
-  mkdir -p $pname/$_pname
-  mkdir -p $pname/tests
-  touch $pname/LICENSE
-  touch $pname/README.md
-  touch $pname/TODO.md
-  touch $pname/setup.py
-  touch $pname/.gitignore
-  touch $pname/install.sh
-  touch $pname/$_pname/__init__.py
-  touch $pname/$_pname/utils.py
-  touch $pname/$_pname/__main__.py
-fi
+[[ -z $1 ]] && help
+
+git_init='false'
+
+while getopts "l:hgp:c:e:" flag; do
+    case "${flag}" in
+        g)
+            git_init='true'
+            ;;
+        l)
+            lang="${OPTARG}"
+            ;;
+        p)
+            project_name="${OPTARG}"
+            ;;
+        c)
+            class_name="${OPTARG}"
+            ;;
+        e)
+            exe_name="${OPTARG}"
+            ;;
+        h)
+            help
+            ;;
+    esac
+done
+
+echo $git_init
+echo $lang
+echo "$project_name"
+echo "$class_name"
```

Comment:
```suggestion
if [[ -z "$lang ]]; then
   .... take input interactively
fi
```
these echoes are redundant as user provides it already. And it doesn't convey much information to user. Can be replaced with interactive user input for generalised options if not present. Context should be more clear in some other comment.

Reply:
Ahh , these were just for testing purpose and I forgot to remove them , sorry for the trouble 🙃🙃

Reply:
Sure